app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.java

```diff
@@ -292,8 +319,41 @@ private void performReadArticleClick() {
     @OnTextChanged(value = R.id.view_description_edit_text,
             callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void pageDescriptionTextChanged() {
+        if (funnel.shouldSeeChecks()) {
+            removeCallbacks(textValidateRunnable);
+            postDelayed(textValidateRunnable, TEXT_VALIDATE_DELAY_MILLIS);
+        } else {
+            isTextValid = true;
+            updateSaveButtonEnabled();
+            setError(null);
+        }
+    }
+
+    void validateText() {
```

Comment:
Should we remove the spaces at the beginning and the end of the text? Users can still send the description if they add a space to the end even if the text contains an invalid format.

Reply:
Sensible.